torch/_inductor/scheduler.py

```diff
@@ -2438,17 +2408,49 @@ def fusable_read_and_write(read: Dep, write: Dep) -> bool:
         # similar to can_inplace, if we are going to fuse a write subsequent to a read
         # require that the indexing and size is the same
         for write in node2.read_writes.writes:
+            if not isinstance(write, MemoryDep):
+                continue
             for read in node1.read_writes.reads:
                 if write.name != self.mutation_renames.get(read.name, read.name):
                     continue
 
                 # bail on StarDep
-                if not fusable_read_and_write(read=read, write=write):
+                if not self.fusable_read_and_write(read, write):
                     why("fusing a write into a read with different indexing formula")
                     return False
 
         return True
 
+    # StarDep doesn't match MemoryDep, different indices don't match
+    # However, broadcasting sometimes strips dimensions, and if that's the case
+    # we still can match unmet dep
+    # if there's indirect indexing, don't match it
+    def fusable_read_and_write(self, read: Dep, write: MemoryDep) -> bool:
+        if isinstance(read, MemoryDep):
+            if read.mode == write.mode and write.mode is not None:
+                return True
+            read_name = read.name
+            if read.name in self.mutation_renames:
+                read_name = self.mutation_renames[read.name]
```

Comment:
NIT: I would expect `get`was cheaper since it only has to lookup in the dict once in the case where `read_name` is in the dict.

Reply:
I would have expected that as well.  `get` is in fact measurably slower.

Reply:
Although it looks like I should have used the local for the `if` instead of re-reading the `read.name`...